data_model/src/lib.rs

```diff
@@ -720,7 +712,20 @@ where
 /// and `PartialCmp` implementations.
 pub trait Identifiable: Debug {
     /// The type of the `Id` of the entity.
-    type Id: Into<IdBox> + fmt::Display + fmt::Debug + Clone + Eq + Ord;
+    type Id: Into<IdBox>
+        + fmt::Display
+        + fmt::Debug
+        + Clone
+        + PartialEq
+        + Eq
+        + PartialOrd
+        + Ord
+        + core::hash::Hash
+        + Decode
+        + Encode
+        // + Deserialize<'a>
+        + Serialize
+        + IntoSchema;
```

Comment:
here

Reply:
`PartialEq` and `PartialOrd` are implied

Reply:
I wouldn't place `IntoSchema` bound on the type if we can do without it. It's somewhat of an internal detail

Reply:
https://github.com/hyperledger/iroha/pull/2497#discussion_r923307138

Reply:
I would say that one should specify bounds on associated types only if the bound is used from within the trait. Otherwise, the the bound requirement is external to the trait and should be controlled from outside with `where` clause. This is similar to the fact that defining bounds on a struct definition is considered a bad practice. Rather you specify bounds on impl blocks.

This is probably the case for all of the bounds on `type Id` here, but I didn't want to insist on it

Reply:
https://github.com/hyperledger/iroha/pull/2497#discussion_r923494428